migration/migrator/migrations/course/20240630190649_course_material.py

```diff
@@ -0,0 +1,45 @@
+"""Migration for a given Submitty course database."""
+
+
+def up(config, database, semester, course):
+    """
+    Run up migration.
+
+    :param config: Object holding configuration details about Submitty
+    :type config: migrator.config.Config
+    :param database: Object for interacting with given database for environment
+    :type database: migrator.db.Database
+    :param semester: Semester of the course being migrated
+    :type semester: str
+    :param course: Code of course being migrated
+    :type course: str
+    """
+    
+    database.execute("ALTER TABLE course_materials ADD COLUMN IF NOT EXISTS uploaded_by character varying(255) REFERENCES users(user_id) DEFAULT NULL;")
+    database.execute("ALTER TABLE course_materials ADD COLUMN IF NOT EXISTS uploaded_date TIMESTAMP WITH TIME ZONE DEFAULT NULL;")
+    database.execute("ALTER TABLE course_materials ADD COLUMN IF NOT EXISTS last_edit_by character varying(255) REFERENCES users(user_id) DEFAULT NULL;")
+    database.execute("ALTER TABLE course_materials ADD COLUMN IF NOT EXISTS last_edit_date TIMESTAMP WITH TIME ZONE DEFAULT NULL;")
+    database.execute("""
+        ALTER TABLE course_materials
+        ADD CONSTRAINT check_dates
+        CHECK (
+            uploaded_date IS NULL 
+            OR last_edit_date IS NULL 
+            OR uploaded_date <= last_edit_date
+        );
+    """)
```

Comment:
```suggestion
database.execute("""
    ALTER TABLE course_materials ADD COLUMN IF NOT EXISTS uploaded_by character varying(255) REFERENCES users(user_id) DEFAULT NULL;
    ALTER TABLE course_materials ADD COLUMN IF NOT EXISTS uploaded_date TIMESTAMP WITH TIME ZONE DEFAULT NULL;
    ALTER TABLE course_materials ADD COLUMN IF NOT EXISTS last_edit_by character varying(255) REFERENCES users(user_id) DEFAULT NULL;
    ALTER TABLE course_materials ADD COLUMN IF NOT EXISTS last_edit_date TIMESTAMP WITH TIME ZONE DEFAULT NULL;
    ALTER TABLE course_materials
    ADD CONSTRAINT check_dates
    CHECK (
        uploaded_date IS NULL 
        OR last_edit_date IS NULL 
        OR uploaded_date <= last_edit_date
    );
""")
```
By wrapping them in a single execute command, you can ensure that all the SQL commands are executed together as a single transaction, and if any of the commands fail, the entire set of commands will not be executed, maintaining the integrity of the database.